File: 3349. Adjacent Increasing Subarrays Detection I/sol.go
package main

func hasIncreasingSubarrays(nums []int, k int) bool {

	if len(nums) < 2*k {
		return false
	}

	for start := 0; start < len(nums)-k; {

		sizeFirst := getMaxSizeOfIncreasing(start, nums)
		if sizeFirst >= 2*k {
			return true
		}

		if sizeFirst >= k {
			sizeSecond := getMaxSizeOfIncreasing(start+sizeFirst, nums)
			if sizeSecond >= k {
				return true
			} else {
				start += sizeFirst + sizeSecond
			}
		} else {
			start += sizeFirst
		}
	}

	return false
}

func getMaxSizeOfIncreasing(start int, nums []int) int {
	for i := 1; i < len(nums)-start; i++ {
		if nums[start+i-1] >= nums[start+i] {
			return i
		}
	}

	return len(nums) - start
}

File: 3349. Adjacent Increasing Subarrays Detection I/sol.c
#include <stdbool.h>

int maxIncreasingLength(int* nums, int numsSize, int start) {
    for (int i = 1; i < numsSize - start; i++) {
        if (nums[start + i - 1] >= nums[start + i])
            return i;
    }
    return numsSize - start;
}

bool hasIncreasingSubarrays(int* nums, int numsSize, int k) {
    if (numsSize < 2 * k)
        return false;
    for (int start = 0; start < numsSize - k;) {
        int firstSize = maxIncreasingLength(nums, numsSize, start);

        if (firstSize >= 2 * k)
            return true;

        if (firstSize < k) {
            start += firstSize;
        } else {
            int secondSize =
                maxIncreasingLength(nums, numsSize, start + firstSize);

            if (secondSize < k) {
                start += firstSize + secondSize;
            } else
                return true;
        }
    }
    return false;
}